lib/repos/remote/remote.go

```diff
@@ -0,0 +1,112 @@
+package remote
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/hofstadter-io/hof/lib/repos/git"
+	"github.com/hofstadter-io/hof/lib/repos/oci"
+	"github.com/hofstadter-io/hof/lib/repos/utils"
+)
+
+// Parse parses a module name and returns
+// the appropriate remote for it.
+func Parse(mod string) (Remote, error) {
+	// TODO: Should pass a context in.
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	switch {
+	case isGit(ctx, mod):
+		return newGitRemote(mod), nil
+	case isOCI(ctx, mod):
+		return ociRemote{mod: mod}, nil
+	}
+
+	return nil, errors.New("remote not known")
+}
+
+type (
+	LocalDir string
```

Comment:
What is the benefit of making these types, and then having to cast strings to them where they are used, liken in `cache/fetch.go`

Reply:
It makes for a cleaner API. I'm hoping that we can eventually refactor other parts of the system to know about these types (so casting won't be necessary).